src/main.js

```diff
@@ -145,6 +147,35 @@ class PercyClient {
     return this._httpPost(`${this.apiUrl}/builds/${buildId}/resources/`, data);
   }
 
+  uploadResources(buildId, contents) {
+    const _this = this;
+    function* generatePromises() {
+      for (const content of contents) {
+        yield _this.uploadResource(buildId, content);
+      }
+    }
+
+    const concurrency = 2;
+    const pool = new PromisePool(generatePromises(), concurrency);
+    return pool.start();
+  }
+
+  uploadMissingResources(buildId, response, resources) {
+    const missingResources = utils.getMissingResources(response);
+    if (!missingResources.length) {
```

Comment:
Potentially drop the `.length`, just do a truthy check here of `if (!missingResources)` since it'll be an empty array.

Reply:
I don't think that will work, `!emptyArray` evaluates to `true` in JS.

Reply:
Sorry, misread that. Will do!

Reply:
Wait nope, take that back haha. Back to my original comment - I don't think that will work as `!emptyArray` evaluates to `false` in JS, so it won't match this `if` condition and will instead batch upload the empty array of missing resources.